lib/ide/parser/src/lib.rs

```diff
@@ -43,13 +62,55 @@ impl Parser {
 
     /// Obtains a default parser implementation, panicking in case of failure.
     pub fn new_or_panic() -> Parser {
-        Parser::new()
-            .unwrap_or_else(|e| panic!("Failed to create a parser: {:?}", e))
+        Parser::new().unwrap_or_else(|e| panic!("Failed to create a parser: {:?}", e))
     }
 }
 
 impl api::IsParser for Parser {
-    fn parse(&mut self, program:String, ids:IdMap) -> api::Result<api::Ast> {
+    fn parse(&mut self, program:String, ids:IdMap) -> api::Result<Ast> {
         self.deref_mut().parse(program,ids)
     }
 }
+
+
+
+// ====================
+// === SharedParser ===
+// ====================
+
+/// Handle to a parser implementation, which keeps the underlying parser in Rc, so this handle can
+/// be shallowly cloned.
+#[derive(Clone,Debug)]
+pub struct SharedParser(Rc<RefCell<dyn api::IsParser>>);
+
+impl SharedParser {
+    /// Obtains a default parser implementation.
+    #[cfg(not(target_arch = "wasm32"))]
+    pub fn new() -> api::Result<SharedParser> {
+        let client = wsclient::Client::new()?;
+        let parser = Rc::new(RefCell::new(client));
+        Ok(SharedParser(parser))
+    }
+
+    /// Obtains a default parser implementation.
+    #[cfg(target_arch = "wasm32")]
+    pub fn new() -> api::Result<SharedParser> {
+        let client = jsclient::Client::new()?;
+        let parser = Rc::new(RefCell::new(client));
+        Ok(SharedParser(parser))
+    }
```

Comment:
Why not just wrap whatever `Parser::new` returns?
Or replace current `Parser` with this `SharedParser` altogether. We don't really have much reason to have "unshareable" parser.